hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java

```diff
@@ -8238,7 +8248,9 @@ else if ( getLoadQueryInfluencers().hasEnabledFetchProfiles() ) {
 							if ( entityMappingType.getSuperMappingType() != null ) {
 								// A joined table group was created by an enabled entity graph or fetch profile,
 								// and it's of an inheritance subtype, so we should apply the discriminator
-								entityMappingType.applyDiscriminator( null, null, actualTableGroup, this );
+								getCurrentClauseStack().push( Clause.FROM );
+								registerEntityNameUsage( actualTableGroup, EntityNameUse.TREAT, entityMappingType.getEntityName() );
+								getCurrentClauseStack().pop();
```

Comment:
This is _ugly_, but it's needed to override the `contextAllowsTreatOrFilterEntityNameUse()` method check if we want to register regular entity name uses for entity graphs / fetch profiles and let the normal pruning do the work when translating.